### PR TITLE
16632 show items on group selection

### PR DESCRIPTION
**Pull request by Corwin-Kh**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/views/layers/POIMapLayer.java [hunk 1]
```diff
@@ -324,10 +325,17 @@ public void fileVisibilityChanged() {
 		this.fileVisibilityChanged = true;
 	}
 
+	public void setAndroidAutoPoints(@Nullable Set<PoiUIFilter> points){
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/views/layers/POIMapLayer.java:328]: setAndroidAutoPoiFilters ?

File: OsmAnd/src/net/osmand/plus/auto/FavoriteGroupsScreen.kt [hunk 2]
```diff
@@ -44,11 +44,13 @@ class FavoriteGroupsScreen(
...
-        for (group in favoriteGroups) {
-            if (collectionSize == contentLimit) {
-                break
-            }
+        if (contentLimit < 2) {
+            return
+        }
+        val favoriteGroupsSize = favoriteGroups.size
+        val limitedFavoriteGroups =
+            favoriteGroups.subList(0, favoriteGroupsSize.coerceAtMost(contentLimit - 2))
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/auto/FavoriteGroupsScreen.kt:52]: Why -2 if other screens -1?

File: OsmAnd/src/net/osmand/plus/auto/FavoritesScreen.java [hunk 3]
```diff
@@ -60,6 +62,14 @@ public FavoritesScreen(
 		this.settingsAction = settingsAction;
 		this.surfaceRenderer = surfaceRenderer;
 		selectedGroup = group;
+		getLifecycle().addObserver(new DefaultLifecycleObserver() {
+			@Override
+			public void onDestroy(@NonNull LifecycleOwner owner) {
+				DefaultLifecycleObserver.super.onDestroy(owner);
+				getApp().getOsmandMap().getMapLayers().getFavouritesLayer().setAndroidAutoFavouritePoints(null);
+				getApp().getOsmandMap().refreshMap();
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/auto/FavoritesScreen.java:70]: Probably we should move this two rows in separate methods in each screen, also add getOsmandMap in base screen class

File: OsmAnd/src/net/osmand/plus/views/layers/MapMarkersLayer.java [hunk 4]
```diff
@@ -396,66 +412,67 @@ public void onDraw(Canvas canvas, RotatedTileBox tileBox, DrawSettings nightMode
...
-						PointI marker31 = NativeUtilities.getPoint31FromLatLon(marker.getLatitude(), marker.getLongitude());
-						PointI center31 = NativeUtilities.get31FromElevatedPixel(mapRenderer, tileBox.getCenterPixelX(), tileBox.getCenterPixelY());
-						if (center31 == null) {
-							continue;
-						}
-						Pair<PointF, PointF> line =
-								NativeUtilities.clipLineInVisibleRect(mapRenderer, tileBox, center31, marker31);
-						if (line == null) {
-							continue;
+		if (!getMapView().isCarView()) {
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/views/layers/MapMarkersLayer.java:415]: use local variable isCarView from above

File: OsmAnd/src/net/osmand/plus/views/layers/FavouritesLayer.java [hunk 5]
```diff
@@ -130,7 +140,20 @@ public void onPrepareBufferImage(Canvas canvas, RotatedTileBox tileBox, DrawSett
 		boolean favoritesChanged = this.favoritesChangedTime != favoritesChangedTime;
 		this.favoritesChangedTime = favoritesChangedTime;
 
-		if (hasMapRenderer()) {
+		if (androidAutoFavouritePoints != null) {
```
Comment (Chumva) [inline OsmAnd/src/net/osmand/plus/views/layers/FavouritesLayer.java:143]: AA check isn`t necessary as in other places?

File: OsmAnd/src/net/osmand/plus/auto/BaseOsmAndAndroidAutoScreen.kt [hunk 6]
```diff
@@ -58,6 +58,6 @@ abstract class BaseOsmAndAndroidAutoScreen(carContext: CarContext) : Screen(carC
     }
 
     companion object {
-        private const val DEFAULT_CONTENT_LIMIT = 12
+        private const val DEFAULT_CONTENT_LIMIT = 100
```
Comment (alex-osm) [inline OsmAnd/src/net/osmand/plus/auto/BaseOsmAndAndroidAutoScreen.kt:61]: Why 100? Possible crash

File: OsmAnd/src/net/osmand/plus/auto/FavoriteGroupsScreen.kt [hunk 7]
```diff
@@ -44,11 +44,13 @@ class FavoriteGroupsScreen(
                 .setBrowsable(true)
                 .setOnClickListener { onClickFavoriteGroup(null) }
                 .build())
-        var collectionSize = 1
-        for (group in favoriteGroups) {
-            if (collectionSize == contentLimit) {
-                break
-            }
+        if (contentLimit < 2) {
```
Comment (alex-osm) [inline OsmAnd/src/net/osmand/plus/auto/FavoriteGroupsScreen.kt:47]: Is it possible?

File: OsmAnd/src/net/osmand/plus/views/layers/FavouritesLayer.java [hunk 8]
```diff
@@ -111,6 +117,10 @@ public void onDraw(Canvas canvas, RotatedTileBox tileBox, DrawSettings settings)
 		}
 	}
 
+	public void setAndroidAutoFavouritePoints(@Nullable List<FavouritePoint> points) {
```
Comment (alex-osm) [inline OsmAnd/src/net/osmand/plus/views/layers/FavouritesLayer.java:120]: Bad approach. Do not inject AA specific code in layers. For example you may introduce some custom data iterator and assign it to layer(s).

File: OsmAnd/src/net/osmand/plus/auto/FavoriteGroupsScreen.kt (hunk 7 above)
Comment (Corwin-Kh) [reply OsmAnd/src/net/osmand/plus/auto/FavoriteGroupsScreen.kt:47]: Technically yes. If ConstraintManager gives such value

File: OsmAnd/src/net/osmand/plus/views/layers/base/OsmandMapLayer.java [hunk 9]
```diff
@@ -83,6 +83,23 @@ public abstract class OsmandMapLayer implements MapRendererViewListener {
 	protected PointI movableObject;
 	protected int pointsOrder = 0;
 
+	public static class CustomMapObjects<T> {
+		protected List<T> customMapObjects;
+
+		@NonNull
+		public List<T> getMapObjects() {
+			if (customMapObjects == null) {
+				return new ArrayList<>();
```
Comment (alex-osm) [inline OsmAnd/src/net/osmand/plus/views/layers/base/OsmandMapLayer.java:92]: Collections.emptyList() ?

File: OsmAnd/src/net/osmand/plus/auto/TracksScreen.kt [hunk 10]
```diff
@@ -25,9 +34,35 @@ class TracksScreen(
     private val trackTab: TrackTab
 ) : BaseOsmAndAndroidAutoScreen(carContext) {
     val gpxDbHelper: GpxDbHelper = app.gpxDbHelper
+    var loadGpxFilesThread: Thread? = null
```
[the remark this answers is not in the document]
Comment (Corwin-Kh) [reply OsmAnd/src/net/osmand/plus/auto/TracksScreen.kt:38]: nullable property must be initialized

**Pull request merged**